go/l2geth-exporter/main.go

```diff
@@ -60,7 +66,35 @@ func main() {
 
 }
 
-func getCTCTotalElements(address string, client *ethclient.Client) {
+func getSCCTotalElements(address string, addressLabel string, client *ethclient.Client) {
+	scc := l1contracts.SCC{
```

Comment:
I'd like these 2 functions to be 1, but I'm not sure how to call the correct contract bindings for the different addresses/types.

Reply:
That makes sense, I'll give it a shot later

Reply:
since `CTC` and `SCC` both expose the same method, `TotalElements(context.Context) (*big.Int, error)`, one solution is to use a `Contract` interface, e.g.
```golang
type Contract interface {
    TotalElements(context.Context) (*big.Int, error)
}
```
to hook it up then we'd need to do the following:
 - replace `address string` argument with `contract Contract` on generic `getTotalElements` method
 - ensure generic method calls `contract.GetTotalElements(ctx)`
 - instantiate the `scc` and `ctc` structs outside of the go routines
 - call `go getTotalElements(scc, ...)` and `go getTotalElements(ctc, ...)`

Reply:
🙌 amazing, exactly what I was looking for